### ISSUE TITLE
[Bug] accelerometer_integral_dt is vibration.

### ISSUE THREAD

**Issue by Abner0907** [PX4/PX4-Autopilot#22197]
### Describe the bug

My code is base on the version of v1.12.3.
I have done some changes in the PX4-Autopilot. But I have not change drivers about the ICM20689.

now I found sometimes the `accelerometer_integral_dt` and `gyro_integral_dt` in the topic of `sensor_comibined` has vibrated as the picture below(the normal is near the 5000, but now it can go to 1250 or 46250 randomly with the same code): show as below: 
my log is : 
https://logs.px4.io/plot_app?log=3299a2a2-f7de-433c-bbf8-c306fb9d8c92
https://logs.px4.io/plot_app?log=c2d89a54-b67d-4e69-827e-ae42856ab1d4.
I am very appreciate that who can give me some help.

@dagar @bresch Do you have the similar issue? I found that you have found a issue here https://github.com/PX4/PX4-Autopilot/issues/17804.

![Screenshot from 2023-10-08 14-38-28](https://github.com/PX4/PX4-Autopilot/assets/64906552/d3c12bef-5868-4ae2-9640-ef2bec3aede3)
![Screenshot from 2023-10-08 14-38-37](https://github.com/PX4/PX4-Autopilot/assets/64906552/150f6198-94f2-4a74-9e93-4dd320f540b7)

### To Reproduce

See error

### Expected behavior

nothing

### Flight Log

https://logs.px4.io/plot_app?log=c2d89a54-b67d-4e69-827e-ae42856ab1d4
https://logs.px4.io/plot_app?log=3299a2a2-f7de-433c-bbf8-c306fb9d8c92

### Software Version

v1.12.3

### Flight controller

Multicopter

### Vehicle type

Multicopter

**Comment by dagar** [comment]
> @dagar @bresch Do you have the similar issue? I found that you have found an issue here https://github.com/PX4/PX4-Autopilot/issues/17804.

No, depending on the board there could be a little bit of jitter due to things like multiple sensors sharing the same bus, but even that should be on the order of tens to hundreds of microseconds (eg ~1% of that integration period). Something is very wrong if you're seeing it jump to ~46 milliseconds periodically.

I'm not sure we'll be able to help all that much because this is custom hardware on older PX4, but with everything running you could grab the output of `sensors status`, `work_queue status`,`uorb top`, `top`, and maybe `perf`. From that we might be able to find the culprit.

**Comment by Abner0907** [comment]
@dagar  thank you for your reply.
I share the outputs as below. 
onething I find the different, the normal is show as below: 
![Screenshot from 2023-10-09 10-49-00](https://github.com/PX4/PX4-Autopilot/assets/64906552/8a49d2f2-9b1a-499d-a2eb-1af91cf93f8a)

the below show the unnormal state: 
![Screenshot from 2023-10-09 10-38-33](https://github.com/PX4/PX4-Autopilot/assets/64906552/80aa4c1d-6b87-4461-afbc-31d72d65118f)
![Screenshot from 2023-10-09 10-39-19](https://github.com/PX4/PX4-Autopilot/assets/64906552/83aa71b6-9399-4d75-9832-a82c0cdcebc0)
![Screenshot from 2023-10-09 10-38-49](https://github.com/PX4/PX4-Autopilot/assets/64906552/2b054340-8f7b-4c52-98d6-999abf268e31)
![Screenshot from 2023-10-09 10-39-32](https://github.com/PX4/PX4-Autopilot/assets/64906552/a5239c62-823c-4c63-b310-20eb347a5a6c)
![Screenshot from 2023-10-09 10-48-02](https://github.com/PX4/PX4-Autopilot/assets/64906552/b1b5acb3-7db0-49f0-8ee9-fbecfd5336ce)

**Comment by Abner0907** [comment]
I doubt the reason is caused by the driver of `ICM20689` or  `vehicle_imu`  or `sensors` thread. 
just record  the relates PR and issues with the similar issue for improvement.
https://github.com/PX4/PX4-Autopilot/issues/17804  (variations in sensor_combined/accelerometer_integral_dt )
https://github.com/PX4/PX4-Autopilot/pull/17805 (drivers/imu/invensense: store offset registers on configure, trigger reset on any change)
https://github.com/PX4/PX4-Autopilot/pull/17817 (sensors/vehicle_imu: minor IMU integration improvements)
https://github.com/PX4/PX4-Autopilot/pull/17818 (unmerge drivers/imu: minor scheduling improvements to tolerate occasional jitter (eg multiple sensors sharing same bus))
https://github.com/PX4/PX4-Autopilot/pull/17862 (drivers/imu: set sample timestamp from interrupt (if available))
